pkg/puppetdb/nodes.go

```diff
@@ -2,15 +2,35 @@ package puppetdb
 
 import (
 	"fmt"
+	"strconv"
 )
 
+func (p Pagination) toParams() map[string]string {
+	params := map[string]string{}
+	if p.Limit > 0 {
+		params["limit"] = strconv.Itoa(p.Limit)
+	}
+	if p.Offset > 0 {
+		params["offset"] = strconv.Itoa(p.Offset)
+	}
+
+	if p.IncludeTotal {
+		params["include_total"] = strconv.FormatBool(p.IncludeTotal)
+	}
+	return params
+}
+
 // Nodes will return all nodes matching the given query. Deactivated and expired nodes aren’t included in the response.
-func (c *Client) Nodes(query string) ([]Node, error) {
+func (c *Client) Nodes(query string, pagination Pagination) ([]Node, error) {
```

Comment:
This looks good, though we should add pagination to the other entities (inventory, facts etc )

Reply:
Happy to leave rolling it out more widely to other PRs. I'd like to make it a `pagination *Pagination` though so that you can pass `nil` when you don't care.